core/src/test/scala/unit/kafka/cluster/PartitionLockTest.scala

```diff
@@ -348,6 +348,7 @@ class PartitionLockTest extends Logging {
     val controllerEpoch = 0
     val replicas = (0 to numReplicaFetchers).map(i => Integer.valueOf(brokerId + i)).toList.asJava
     val isr = replicas
+    replicas.forEach(replicaId => when(metadataCache.getAliveBrokerEpoch(replicaId)).thenReturn(Some(1L)))
```

Comment:
Why do we need this extra code?

Reply:
We use `mock(classOf[MetadataCache])` for `metadataCache`, so it didn't really get into `KRaftMetadataCache` block. In this PR, we remove `match` block, so we have to add related mock.

https://github.com/apache/kafka/blob/94a1bfb1281f06263976b1ba8bba8c5ac5d7f2ce/core/src/test/scala/unit/kafka/cluster/PartitionLockTest.scala#L270

https://github.com/apache/kafka/blob/94a1bfb1281f06263976b1ba8bba8c5ac5d7f2ce/core/src/main/scala/kafka/cluster/Replica.scala#L116-L125